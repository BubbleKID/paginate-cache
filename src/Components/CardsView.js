import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import SimpleCard from "./SimpleCard";
import Pagination from "./Pagination";
import TemporaryDrawer from "./TemporaryDrawer";

import axios from "axios";
const styles = theme => ({
  root: {
    flexGrow: 1,
    paddingTop: theme.spacing.unit * 5.5,
    paddingRight: theme.spacing.unit * 30,
    paddingBottom: theme.spacing.unit * 3,
    paddingLeft: theme.spacing.unit * 30,
    backgroundColor: "#eeeeee"
  },
  control: {
    padding: theme.spacing.unit * 3
  }
});
class CardsView extends Component {
  constructor(props) {
    super(props);
    this.state = {
      //token: "",
      cardsPerPage: 12,
      currPage: 1,
      currCardData: "",
      drawerOpen: false
    };
    this.handleOpenDrawer = this.handleOpenDrawer.bind(this);
  }
  handleOpenDrawer(value) {
    this.setState({
      currCardData: value,
      drawerOpen: true
    });
  }

  componentDidMount() {
    let self = this;
    axios
      .post(
        "https://atr-test-dh1.aiam-dh.com/atr-gateway/identity-management/api/v1/auth/token",
        {
          password: "password",
          username: "frontendtest"
        }
      )
      .then(function(response) {
        //console.log(response.data.token);
        // self.setState({
        //   token: response.data.token
        // });
        self.props.dispatch({ type: "SET_TOKEN", token: response.data.token });
        axios
          .get(
            "https://atr-test-dh1.aiam-dh.com/atr-gateway/ticket-management/api/v1/tickets?ticketType=incident&sortDirection=DESC&page=0&perPage=12",
            {
              headers: {
                apiToken: self.props.token,
                "Content-Type": "application/json",
                Accept: "application/json"
              }
            }
          )
          .then(function(response) {
            const totalPage =
              Math.ceil(
                parseInt(response.headers["x-total-count"]) /
                  self.state.cardsPerPage
              ) - 1;
            self.props.dispatch({
              type: "SET_CARDS",
              cards: response.data,
              totalPage: totalPage
            });
          })
          .catch(function(error) {
            console.log(error);
          });
      })
      .catch(function(error) {
        console.log(error);
      });
  }

  render() {
    const { classes } = this.props;
    return (
      <React.Fragment>
        {this.state.currCardData.coreData && (
          <TemporaryDrawer
            coreData={this.state.currCardData.coreData}
            serviceData={this.state.currCardData.serviceData}
            drawerOpen={this.state.drawerOpen}
          />
        )}
        <Grid container className={classes.root} justify="center">
          <Grid item>
            <Grid
              container
              className={classes.demo}
              justify="center"
              spacing={16}
            >
              {this.props.cards.map(card => (
                <Grid key={card.coreData.id} item>
                  <SimpleCard
                    card={card}
                    handleOpenDrawer={this.handleOpenDrawer}
                  />
                </Grid>
              ))}
            </Grid>
          </Grid>
          <Pagination
            totalPage={this.props.totalPage}
            currPage={this.state.currPage}
          />
        </Grid>
      </React.Fragment>
    );
  }
}

const mapStateToProps = state => ({
  cards: state.cards,
  totalPage: state.totalPage,
  token: state.token
});

CardsView.propTypes = {
  classes: PropTypes.object.isRequired
};

export default connect(mapStateToProps)(withStyles(styles)(CardsView));
