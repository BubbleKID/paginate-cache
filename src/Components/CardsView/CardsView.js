import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import SimpleCard from "../SimpleCard/SimpleCard";
import Pagination from "../Pagination/Pagination";
import * as actions from "./CardsViewActions";

const styles = theme => ({
  root: {
    paddingTop: theme.spacing.unit * 5.5,
    paddingRight: theme.spacing.unit * 30,
    paddingBottom: theme.spacing.unit * 3,
    paddingLeft: theme.spacing.unit * 30,
    backgroundColor: "#eeeeee"
  },
  control: {
    padding: theme.spacing.unit * 3
  }
});

class CardsView extends Component {
  componentDidMount() {
    this.props.dispatch(actions.fetchCards());
  }

  render() {
    const { classes } = this.props;
    return (
      <React.Fragment>
        <Grid container className={classes.root} justify="center">
          <Grid item>
            <Grid
              container
              className={classes.cards}
              justify="center"
              spacing={16}
            >
              {this.props.cards.map(card => (
                <Grid key={card.coreData.id} item xs={3}>
                  <SimpleCard card={card} />
                </Grid>
              ))}
            </Grid>
          </Grid>
          <Pagination
            totalPage={this.props.totalPage}
            currPage={this.props.currPage}
          />
          
         
        </Grid>
      </React.Fragment>
    );
  }
}

const mapStateToProps = state => ({
  cards: state.cards,
  totalPage: state.totalPage,
  currPage: state.currPage
});

CardsView.propTypes = {
  classes: PropTypes.object.isRequired
};

export default connect(mapStateToProps)(withStyles(styles)(CardsView));
