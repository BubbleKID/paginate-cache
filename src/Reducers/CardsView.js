const innitialState = {
  currPage: 1,
  cards: [],
  totalPage: 0,
  currCardData: {}
};

function CardsViewReducer(state = innitialState, action) {
  switch (action.type) {
    case "SET_CARDS":
      return {
        ...state,
        cards: action.cards,
        totalPage: action.totalPage
      };
    case "SET_CURRENTCARDS":
      return {
        ...state,
        currCardData: action.currCardData
      };
    default:
      return state;
  }
}

export default CardsViewReducer;
