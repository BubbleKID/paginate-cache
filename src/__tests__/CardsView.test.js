import React from "react";
import { shallow } from "enzyme";
import CardsView from "../Components/CardsView";
describe("CardsView", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<CardsView debug />);
    expect(component).toMatchSnapshot();
  });
});
