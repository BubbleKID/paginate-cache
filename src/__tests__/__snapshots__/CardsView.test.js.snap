// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsView should render correctly in "debug" mode 1`] = `
<CardsView
  classes={
    Object {
      "control": "CardsView-control-2",
      "root": "CardsView-root-1",
    }
  }
  debug={true}
/>
`;
